### ISSUE TITLE
Opening the Run Configurations Dialog Freezes UI

### ISSUE THREAD

**Issue by HeikoKlare** [vi-eclipse/Eclipse-Platform#3]
### Current Behavior

Opening the "Run Configurations" dialog indeterministically takes a long time, in particular when first opened after starting Eclipse. It freezes the UI for a multitude of seconds.

EDIT: this behavior is due to several bad implementations that contribute to an overall bad performance. This issue will then serve as an umbrella for such smaller issues.

### Expected Behavior

The "Run Configurations" dialog opens fast or shows some progress dialog to indicate what is taking so much time to open it. If the operation is still too slow and can NOT be optimized, the operation should be cancelable and the UI should react properly. 

The specific solutions applied for each specific issue should be documented in the issue itself.

Further ideas (they should be addressed in the specific issues) can be found in this comment: https://github.com/vi-eclipse/Eclipse-JDT/issues/8#issuecomment-1761711223. Shortly summarized:
* Asynchronous calculation of hierarchies (for the JUnit configs)
* Disabling the UI properly while things are being calculated

### Tasks
* https://github.com/vi-eclipse/Eclipse-PDE/issues/2
* https://github.com/eclipse-jdt/eclipse.jdt.ui/pull/675
* https://github.com/vi-eclipse/Eclipse-PDE/issues/3
* Options for further tasks, see https://github.com/vi-eclipse/Eclipse-Platform/issues/3#issuecomment-2075147987

**Comment by fedejeanne** [comment]
See https://github.com/eclipse-jdt/eclipse.jdt.ui/issues/653

**Comment by HeikoKlare** [comment]
Just not to forget, see https://github.com/vi-eclipse/Eclipse-JDT/issues/8#issuecomment-1761711223 for further ideas.
They should still be properly documented in this issue.

**Comment by fedejeanne** [comment]
> Just not to forget, see https://github.com/vi-eclipse/Eclipse-JDT/issues/8#issuecomment-1761711223 for further ideas.
They should still be properly documented in this issue.

Looking back at this issue, I think we should stick to simpler solutions first like moving the long-running operations to the activation of the tabs (https://github.com/vi-eclipse/Eclipse-PDE/issues/2), adding progress indicators (https://github.com/eclipse-jdt/eclipse.jdt.ui/pull/675) and -if necessary- making the operation cancelable (https://github.com/vi-eclipse/Eclipse-PDE/issues/3).

@HeikoKlare what do you say if we keep this issue as an umbrella for all other smaller issues? If you agree to it, I can change the description of this issue and start listing the known issues and PRs related to it and we can also add new specific issues later.

I also briefly tested the performance of the _Launch Configurations Dialog_ in our PV workspace and is looking fine so I don't have any specific issues/proposals to add to this umbrella issue right now.

**Comment by HeikoKlare** [comment]
> what do you say if we keep this issue as an umbrella for all other smaller issues?

Yes, this should serve as an umbrella issue. Seems like I accidentally moved it to the wrong backlog category, as it had been placed under "Aggregates" and is not part of "Ready: Atomic" instead of "Ready: Epics".
Still the ideas documented in https://github.com/vi-eclipse/Eclipse-JDT/issues/8#issuecomment-1761711223 should be documented either here or in an atomic issue under the umbrella of this, even though they will have low priority and may not be addressed soon.

**Comment by fedejeanne** [comment]
> Yes, this should serve as an umbrella issue

👍 

> Still the ideas documented in https://github.com/vi-eclipse/Eclipse-JDT/issues/8#issuecomment-1761711223 should be documented either here or in an atomic issue under the umbrella of this, even though they will have low priority and may not be addressed soon.

Ok. I currently have no proposals though, since the 2 "big" issues (calculation of the class hierarchies for JUnit configs and Tracing Options for the PDE configs) have been addressed and the solution is good enough.

**Comment by fedejeanne** [comment]
Looking at it in our Grafana, this issue doesn't happen very often (5 times in the past 90 days) but it is still present:

<img width="1647" height="446" alt="Image" src="https://github.com/user-attachments/assets/a035f993-8d19-4156-a756-af58db3b2e97" />

---

I also added it to the dashboard "_Usual suspects (UI freezes)_" under the name _LaunchConfigurationDialog_

**Comment by HeikoKlare** [comment]
Closing this as we further track progess and prioritize this issue in an internal backlog.
